src/odb/src/db/dbPowerSwitch.cpp

```diff
@@ -240,16 +245,115 @@ void dbPowerSwitch::addOnState(const std::string& on_state)
   obj->_on_state.push_back(on_state);
 }
 
+void dbPowerSwitch::setLibCell(dbMaster* master)
+{
+  _dbPowerSwitch* obj = (_dbPowerSwitch*) this;
+  obj->_lib_cell = master->getImpl()->getOID();
+  obj->_lib = master->getLib()->getImpl()->getOID();
+}
+
+void dbPowerSwitch::addPortMap(const std::string& model_port,
+                               const std::string& switch_port)
+{
+  _dbPowerSwitch* obj = (_dbPowerSwitch*) this;
+  dbMaster* master = (dbMaster*) dbMaster::getMaster(
+      dbLib::getLib((dbDatabase*) obj->getImpl()->getDatabase(), obj->_lib),
+      obj->_lib_cell);
+
+  if (!master) {
+    obj->getImpl()->getLogger()->error(
+        utl::ODB,
+        32001,
+        "Cannot map port {} to {} because no lib cell is added",
+        model_port,
+        switch_port);
+    return;
+  }
+
+  dbMTerm* mterm = master->findMTerm(switch_port.c_str());
```

Comment:
what if mterm is null - that should be an error.

Reply:
fixed